src/cargo/core/compiler/mod.rs

```diff
@@ -1081,8 +1101,9 @@ fn on_stderr_line(
     package_id: PackageId,
     target: &Target,
     options: &mut OutputOptions,
+    rustc_client: Option<&Client>,
```

Comment:
I think it may perhaps be a bit cleaner to instead of passing `rustc_client` around and sending it through messages to instead store it elsewhere. The processes spawned by the job queue I think should all have an optional `Client` associated with them (stored internally in `ProcessBuilder`), and `job_queue` could keep track of a map from id to `Client` in that case.